src/mnemonics/electrum-words.h

```diff
@@ -103,6 +112,16 @@ namespace crypto
     bool bytes_to_words(const crypto::secret_key& src, std::string& words,
       const std::string &language_name);
 
+    /*!
+     * \brief Converts bytes (secret key) to seed words.
+     * \param  src           16-byte legacy secret key
+     * \param  words         Space delimited concatenated words get written here.
+     * \param  language_name Seed language name
+     * \return               true if successful false if not. Unsuccessful if wrong key size.
+     */
+    bool bytes_to_words(const crypto::legacy16B_secret_key& src, std::string& words, 
+        const std::string &language_name);   
```

Comment:
Do we really need this? I think we're primarily interested in being able to _read_ the 13-word seed, not write.

Reply:
If you wanted to support 16 byte seeds then yes you do because you need to be able to display the 13 word seed as you cannot take a 16 byte seed and accurately turn it into 25 words for your users.

Reply:
I guess the definition of "support" is different between you and me. What I had in mind is to be able to turn a 13-word seed into a pair of spend & view secret keys. Note that this makes the wallet keys "non-deterministic" because the relation `viewkey == reduce32(hash(spendkey))` doesn't hold anymore. Therefore, the wallet can't print the 25-word seed, which I think (and @moneromooo-monero also seems to agree) is fine. The user can just keep that 13-word seed, or alternatively keep the spend & view secret keys.

If instead we are to enable printing the 13-word seed, the wallet needs to store that 16-byte data separately. I.e., the relationships are now `spendkey == reduce32(seed16)` and `viewkey == reduce32(hash(seed16))`, so the 16-byte seed data must be preserved in the wallet keys file in order to be able to print the 13-word seed. This can be of course implemented, but will involve quite a bit of additional code. I'm rather against that.

If the former approach is to be taken, this function `bytes_to_words` is unnecessary.

Reply:
In MyMonero when the user enters their mnemonic we store the seed (under encryption), then if the user requests the mnemonic is displayed we have to have some way to convert the seed to the correct mnemonic. Without that requirement yeah not necessary :) But I'm sure people will want to use that function even if the CLI app doesn't need it. If libmonero is to be able to be considered for embedding in exchanges, wallets, and other utility code, I don't think we can limit the scope of use-cases to simplewallet. (Else I dont think they would have any need to exist as simplewallet is quite good at its use case. )